src/Storage/Storage.Management/File/NewAzureStorageShare.cs

```diff
@@ -112,11 +112,32 @@ public string AccessTier
 
         [Parameter(Mandatory = false, HelpMessage = "Create a snapshot of existing share with same name.")]
         public SwitchParameter Snapshot { get; set; }
+        
+        [Parameter(Mandatory = false,
+            HelpMessage = "Sets protocols for file shares. It cannot be changed after file share creation. Possible values include: 'SMB', 'NFS'")]
+        [ValidateSet(EnabledProtocols.NFS,
+            EnabledProtocols.SMB,
+            IgnoreCase = true)]
+        public string EnabledProtocol { get; set; }
```

Comment:
if you very unlikely to extend enabled protocols anymore, it is ok to use validate set. If not, argument completer will be better.

Reply:
We don't want user to input value except the NFS/SMB here. It seems argument completer will still allow user to input other values. So it seems ValidateSet will be more suitable here.

BTW, if we add more value to the validationSet, it should not be breaking, right? (I have done such change before and not taken as breaking.)